source/lib.js

```diff
@@ -129,6 +183,26 @@ function hasYield(exp) {
   return gatherRecursiveWithinFunction(exp, ({ type }) => type === "Yield").length > 0
 }
 
+function hoistRefDecs(statements) {
+  gatherRecursiveAll(statements, (s) => s.hoistableDec)
+    .forEach(node => {
+      const { hoistableDec } = node
+
+      // TODO: expand set to include other parents that can have hoistable decs attached
```

Comment:
Yeah, I worry that currently a declaration in an expression that is deeply inside an `if` body (with a function declaration or various other things in between) will end up attaching to the `if` here. But reading below, I guess that won't happen with the current grammar.  In the future I guess this is where `Statement` AST nodes would be helpful, and would enable declarations in more general places in expressions.

Reply:
👍 I imagine this will get refined a bit as we fill in more refs and do more with blocks and hoisting.